oauth2_provider/oauth2_validators.py

```diff
@@ -332,15 +332,14 @@ def _get_token_from_authentication_server(
             scope = content.get("scope", "")
             expires = make_aware(expires)
 
-            access_token, _created = AccessToken\
-                .objects.select_related("application", "user")\
-                .update_or_create(token=token,
-                                  defaults={
-                                      "user": user,
-                                      "application": None,
-                                      "scope": scope,
-                                      "expires": expires,
-                                  })
+            access_token, _created = AccessToken.objects.update_or_create(
```

Comment:
why select_related is removed?

Reply:
`update_or_create` uses `select_for_update` in the background if model exists. It's not supported to use `select_for_update` with nullable relations as mentioned in [Django docs](https://docs.djangoproject.com/en/2.2/ref/models/querysets/#select-for-update).

So, I could change the line to this:

```Python
access_token, _created = AccessToken.objects.select_related().update_or_create(...)
```

But calling `select_related` with no arguments only follows non nullable relations, nullable relations must be specified (see [docs](https://docs.djangoproject.com/en/2.1/ref/models/querysets/#select-related)). Since `user` and `application` are nullable relations, there is no point to call `select_related` with no arguments, and we can't specify this fields because of `update_or_create`.